### PR TITLE
Revising text on include files and verification

### LINKED ISSUE #3310: Is it OK to verify related files independently?
### What change in documentation do you suggest?

Is verifying N files with a single `dafny` call equivalent to verifying them separately?
Or is there something to be gained by verifying all related files together? 

Asking this because verifying files independently is very convenient for a build system, and is the default behavior anyway. However this is not explicitly stated in the documentation.

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #3310

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/UserGuide.md [hunk 1]
```diff
@@ -171,6 +166,10 @@ A few options are not part of a command. In these cases any single-hyphen spelli
 
 The `dafny resolve` command checks the command-line and then parses and typechecks the given files and any included files.
 
+Note that a complete program must be presented to the dafny tool, either by listing all files on the command-line, or by using `include` directives,
+of by some combination. An incomplete program will likely result in errors because of unresolved names. A program need not include all implementations
+of all functions in order to verify parts of it, but will need all implementations in order to compile a working executable.
```
Comment (MikaelMayer) [inline docs/DafnyRef/UserGuide.md:171]: How is it possible to provide a declaration à la c-header and an implementation on anther file? I don't see how.

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
Looks good !

File: docs/DafnyRef/UserGuide.md [hunk 2]
```diff
@@ -179,23 +178,39 @@ The options relevant to this command are
...
    - `--track-print-effects`
    - `--warn-shadowing`
    - `--warn-missing-constructor-parentheses`
 
 
 #### 25.5.1.3. `dafny verify` {#sec-dafny-verify}
 
-The `dafny verify` command performs the [`dafny resolve`](#sec-dafny-resolve) checks and then attempts to verify each method in the files listed on the command line. Although the Dafny program being considered
-consists of the listed files and any included files (recursively), by default only listed files are verified.
+The `dafny verify` command performs the [`dafny resolve`](#sec-dafny-resolve) checks and then attempts to verify each method in the program.
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:193]: Fixed

File: docs/DafnyRef/UserGuide.md (hunk 1 above)
Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:171]: There is no analog of c-headers of course, but one can omit implementations of functions for verification, but must have them for compilation.

**Review by MikaelMayer — APPROVED** [review]

File: docs/DafnyRef/UserGuide.md (hunk 1 above)
Comment (hmijail) [reply docs/DafnyRef/UserGuide.md:171]: I was wondering the same and this answer was helpful, so maybe it could be clearer?

Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:171]: OK. I'll clarify in a future PR (since this one is merged) keeping in mind programmers coming from C.

**Pull request merged**